sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidLogicalValuesRule.java

```diff
@@ -93,24 +94,28 @@ public void onMatch(RelOptRuleCall call)
    *
    * @throws IllegalArgumentException for unsupported types
    */
+  @Nullable
   @VisibleForTesting
   static Object getValueFromLiteral(RexLiteral literal, PlannerContext plannerContext)
   {
+    if (RexLiteral.value(literal) == null) {
```

Comment:
BTW I looked at internal calcite code. If literal.getType().getSqlTypeName() is `Boolean` and literal.getSqlTypeName() is not `Boolean` then this method was earlier returning 0 but now will return null.

Reply:
I actually wrote a query that will generate different results after this change 
```
SELECT cast(NULL as boolean)
```
Before
```
[false]
```
After
```
[null]
```

Verified on my local box.

Reply:
I feel that we should let calcite do its thing and not replicate its internal logic as much as possible.

Reply:
Good catch, I checked and as per your comment this is an edge case that changes with this `if` condition.
More specifically, iff `literal.getType.getSqlTypeName() == BOOELAN`, `literal.getSqlTypeName() =/= BOOLEAN` and `literal.value == null`, then we would exit early from the code and return null, but the older method should be returning `false`.

Reply:
er, `CAST(null as boolean)` should be null i think...

Reply:
Abhishek shared a snippet in Calcite's library: [RexUtil](https://github.com/apache/calcite/blob/82dd78a14f6aef2eeec2f9c94978d04b4acc5359/core/src/main/java/org/apache/calcite/sql/SqlUtil.java#L177) which states that it is intended for it to be null.

Reply:
http://sqlfiddle.com/#!9/7fff3f/35

looks like it is another place to use `useStrictBoolean` flag? :) But my larger point was around the internal calcite quirks that we might not be aware of. I am concerned that we will accidentally change the behaviour of existing queries as happened here. I will suggest keeping the scope of this PR to just fixing the bug.

Reply:
We do need the null check since that is getting activated in the test cases. Maybe it can be put in the respective switch case?